qiskit/synthesis/two_qubit/__init__.py

```diff
@@ -12,4 +12,20 @@
 
 """Module containing two-qubit unitary synthesis methods."""
 
-from .two_qubit_decompose import TwoQubitBasisDecomposer, two_qubit_cnot_decompose
+from .two_qubit_decompose import (
+    TwoQubitBasisDecomposer,
+    two_qubit_cnot_decompose,
+    TwoQubitWeylDecomposition,
```

Comment:
A search through `qiskit-community` showed that only these two objects are used. I'd suggest we use this as criterion and only add those classes to the API docs that are really needed 🙂

Reply:
I'm fine starting small, also looking at the code for all the specialized `TwoQubitWeyl*` variants it looks like they're all used internally by `__new__` when you create a new `TwoQubitWeylDecomposition` object.

Reply:
`TwoQubitWeylDecomposer` _kind of_ implies that all the family classes are public in the sense of going to a lot of trouble to make a `repr` that you can `eval` back in (including pickling then b64'ing a Numpy array and making sure `__new__` eats keywords that it throws in the `repr` just to be pretty) (it's a _very_ busy constructor).  That said, I don't really think it's necessary to make those public at all - they're largely not meant to be accessed directly, I think, because of the abstract factory class.